Add function for parsing age identity

cezarmathe opened the pull request:
This new function will look for a line in the input string that starts
with `AGE-SECRET-KEY-1`. That line is then cleaned of whitespaces and
passed to `age.ParseX25519Identity`.

Closes #79.
Fixes #80.

slok commented on internal/key/age/age.go:
```diff
@@ -174,3 +175,22 @@ func askPasswordStdin(r io.Reader, logger log.Logger) func() ([]byte, error) {
 	}
 }
+
+func parseAgeIdentityFunc(d string) (age.Identity, error) {
+	lines := strings.Split(d, "\n")
+
+	var privateKey string = ""
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "AGE-SECRET-KEY-1") {
+			privateKey = strings.Trim(line, " ")
+			break
+		}
+	}
+
+	if privateKey == "" {
+		return nil, fmt.Errorf("no private key found")
+	}
+
+	return age.ParseX25519Identity(privateKey)
+}
```
What do you think if we simplify this by using regexes and `Trimspace`? Something similar to this:

```go
var agePrivKeyRegex = regexp.MustCompile("(?m)(^AGE-SECRET-KEY-.*$)")

func parseAgeIdentityFunc() func(s string) (age.Identity, error) {
	return func(d string) (age.Identity, error) {
		d = agePrivKeyRegex.FindString(d)
		d = strings.TrimSpace(d)

		return age.ParseX25519Identity(d)
	}
}

```

slok commented on tests/manual/keys-priv/key7:
```diff
@@ -1 +1,8 @@
-AGE-SECRET-KEY-169UTU9G4YWRWEW5KTAJEACLHZR2G8TFMW8AUE0S4NRH9R3ZNR66QJW4Y56
\ No newline at end of file
+# This is a comment
```
Could you please add this kind of test to the unit tests instead of the manual ones?
Here are the unit tests: https://github.com/slok/agebox/blob/62fc6726188b4a98de4b9f1e3e7a833e0a6bc72b/internal/key/age/age_test.go#L131-L134

slok reviewed (commented):
Thanks for the PR! I left you some comments :)

cezarmathe commented:
Hi! I've added two more unit tests. Also, I've updated the specific function. I'm not sure if just finding the key in the string is okay. IMO loading a key should fail if it contains anything else besides comments and the actual key. I haven't found a spec for the private key file.

slok commented:
> IMO loading a key should fail if it contains anything else besides comments and the actual key. I haven't found a spec for the private key file.

`age.ParseX25519Identity(privateKey)` will fail anyway whatever you pass if is not correct. Anyway, on your implementation, you only check until the line you have a valid key so you are already ignoring everything below that line that can be anything. Also, you can have new lines between comments that I don't see wrong either and this implementation would fail.

So, taking both ideas and merging them, I think that a good implementation to this would be removing all the comments and then trim:

```go
var removeCommentRegexp = regexp.MustCompile("(?m)(^#.*$)")

func parseAgePrivateFunc() func(s string) (age.Identity, error) {
	return func(d string) (age.Identity, error) {
		d = removeCommentRegexp.ReplaceAllString(d, "")
		d = strings.TrimSpace(d)

		return age.ParseX25519Identity(d)
	}
}
```

What do you think?

slok reviewed (approved):
Awesome, thanks!

slok commented:
New release with this fix: [`v0.5.1`](https://github.com/slok/agebox/releases/tag/v0.5.1)
